### ISSUE TITLE
Move Softmax calculation to tested utility function

### ISSUE THREAD

**Issue by luckylwk** [AI4Finance-Foundation/FinRL#220]
Also align with implementations like tensorflow.

Working on a PR for this.
